src/signal-handling.c

```diff
@@ -129,6 +129,35 @@ static size_t jl_safe_read_mem(const volatile char *ptr, char *out, size_t len)
     return i;
 }
 
+static double profile_autostop_time = -1.0;
+static double profile_peek_duration = 1.0; // seconds
+
+double jl_get_profile_peek_duration(void) {
+    return profile_peek_duration;
+}
+void jl_set_profile_peek_duration(double t) {
+    profile_peek_duration = t;
+    return;
+}
+
+uint64_t profile_show_peek_cond_loc;
+JL_DLLEXPORT void jl_set_peek_cond(uint64_t cond)
+{
+    profile_show_peek_cond_loc = cond;
+    return;
+}
+
+static void jl_check_profile_autostop(void) {
+    if ((profile_autostop_time != -1.0) && (jl_hrtime() > profile_autostop_time)) {
+        profile_autostop_time = -1.0;
+        jl_profile_stop_timer();
+        jl_safe_printf("\n==============================================================\n");
+        jl_safe_printf("Profile collected. A report will print at the next yield point\n");
+        jl_safe_printf("==============================================================\n\n");
+        uv_async_send(profile_show_peek_cond_loc);
```

Comment:
I can't figure out how to pass the `PROF_PEEK_COND[].handle` through here without causing this warning:
```
    CC src/signal-handling.o
../julia/src/signal-handling.c:157:23: warning: incompatible integer to pointer conversion passing 'uint64_t' (aka 'unsigned long long') to parameter of type 'uv_async_t *' (aka 'struct uv_async_s *') [-Wint-conversion]
        uv_async_send(profile_show_peek_cond_loc);
                      ^~~~~~~~~~~~~~~~~~~~~~~~~~
../julia/usr/include/uv.h:923:41: note: passing argument to parameter 'async' here
UV_EXTERN int uv_async_send(uv_async_t* async);
                                        ^
1 warning generated.
```